src/components/SortableTable.js

```diff
@@ -1,84 +1,94 @@
-import React, { useMemo, useState } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'reactstrap';
 
-import { Tooltip } from 'reactstrap';
+import HeaderCell from 'components/HeaderCell';
+import useSortableData from 'hooks/UseSortableData';
 
-import PagedTable from './PagedTable';
-
-const Header = ({ header, headerProps, active, sorts, setSort }) => {
-  const [tooltipOpen, setTooltipOpen] = useState(false);
-
-  const toggleTooltip = () => setTooltipOpen((open) => !open);
-
-  const { tooltip, ...rest } = headerProps;
-  const sortable = !!sorts[header];
-  const tooltipElement = tooltip && (
-    <Tooltip
-      placement="top"
-      boundariesElement="window"
-      trigger="hover"
-      target={header.replace(' ', '')}
-      isOpen={tooltipOpen}
-      toggle={toggleTooltip}
-    >
-      {tooltip}
-    </Tooltip>
-  );
-  if (sortable) {
-    return (
-      <th onClick={() => setSort(header)} scope="col" {...rest} style={{ cursor: 'pointer', ...rest.style }}>
-        <span id={header.replace(' ', '')}>
-          {header}
-          {active ? ' ▼' : ''}
-        </span>
-        {tooltipElement}
-      </th>
-    );
-  } else {
-    return (
-      <th scope="col" {...rest}>
-        <span id={header.replace(' ', '')}>{header}</span>
-        {tooltipElement}
-      </th>
-    );
+export const valueRenderer = (value) => {
+  if (!Number.isFinite(value) || Number.isInteger(value)) {
+    return value;
   }
+  return value.toFixed(2);
 };
 
-const SortableTable = ({ sorts, defaultSort, headers, data, rowF, ...props }) => {
-  const [sort, setSort] = useState(defaultSort);
-  const sortKeyF = sorts[sort];
-  let sortedData;
-  if (Array.isArray(data)) {
-    sortedData = data;
-    if (sortKeyF) {
-      sortedData = useMemo(() => {
-        const result = [...data];
-        result.sort((x, y) => sortKeyF(x) - sortKeyF(y));
-        return result;
-      }, [data, sortKeyF]);
-    }
-  } else {
-    sortedData = data[sort];
-  }
+export const compareStrings = (a, b) => a?.toString?.()?.localeCompare?.(b?.toString?.());
+
+export const SortableTable = ({ data, defaultSortConfig, sortFns, columnProps, totalRow, totalCol, ...props }) => {
+  const { items, requestSort, sortConfig } = useSortableData(data, defaultSortConfig, sortFns);
 
-  const rows = sortedData.map(rowF);
   return (
-    <PagedTable rows={rows} {...props}>
```

Comment:
just cutting pagination entirely?

Reply:
This component was unused before, adding pagination would've changed the behavior of the other components. Effectively I deleted this for being unused than wrote a new component with the same name because it's the most descriptive.